apps/emqx_connector/src/schema/emqx_connector_schema.erl

```diff
@@ -203,11 +203,21 @@ transform_old_style_bridges_to_connector_and_actions_of_type(
                 [<<"bridges">>, to_bin(BridgeType), BridgeName],
                 RawConfigSoFar1
             ),
+            %% Take fields that should be in the top level of the action map
+            TopLevelFields = [<<"resource_opts">>, <<"enable">>, <<"connector">>],
+            TopLevelActionFields = maps:with(TopLevelFields, ActionMap),
+            ParametersActionFields = maps:without(TopLevelFields, ActionMap),
+            %% Action map should be wrapped under parameters key
+            WrappedParameters = #{<<"parameters">> => ParametersActionFields},
+            FinalActionMap = maps:merge(TopLevelActionFields, WrappedParameters),
+            FixedActionMap = emqx_action_info:bridge_v1_to_action_fixup(
+                BridgeType, FinalActionMap
+            ),
```

Comment:
I think I see why the fixup code in Kafka/AEH expects a `kafka` key under `parameters`, which shouldn't happen according to the schema.  Which is a bit confusing.

Couldn't we call the `bridge_v1_to_action_fixup` first, and do the automatic heuristic **only** if the callback is undefined?  That way, all the massaging needed is handled by the node and is closely tied to the v1/action schemas rather than an intermediate step like the wrapping here introduces.